src/coinjoin/context.cpp

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2023 The Dash Core developers
+// Distributed under the MIT/X11 software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <coinjoin/context.h>
+
+#include <net.h>
+#include <policy/fees.h>
+#include <txmempool.h>
+
+#ifdef ENABLE_WALLET
+#include <coinjoin/client.h>
+#endif // ENABLE_WALLET
+#include <coinjoin/server.h>
+
+CJContext::CJContext(CConnman& connman, CTxMemPool& mempool, const CMasternodeSync& mn_sync, bool relay_txes) :
+#ifdef ENABLE_WALLET
+    clientman {
+        [&]() -> CJClientManager* const {
+            assert(::coinJoinClientManagers == nullptr);
+            ::coinJoinClientManagers = std::make_unique<CJClientManager>(connman, mempool, mn_sync, queueman);
+            return ::coinJoinClientManagers.get();
+        }()
+    },
+    queueman {
+        [&]() -> const std::unique_ptr<CCoinJoinClientQueueManager> {
```

Comment:
Lambda function is not needed here, can be simplified to ternary operator.

```
queueman{relay_txes ? std::make_unique<CCoinJoinClientQueueManager>(connman, *clientman, mn_sync) : nullptr}
```

`clientman` requires lambda function due to extra `assert`, no need to change anything.

Reply:
Resolved in latest push